generator/src/main/java/com/scottlogic/deg/generator/restrictions/MergeResult.java

```diff
@@ -1,10 +1,17 @@
 package com.scottlogic.deg.generator.restrictions;
 
 public class MergeResult<T> {
+
+    private static final MergeResult<?> UNSUCCESSFUL = new MergeResult<>();
+
     public final T restrictions;
     public final boolean successful;
 
-    public final static MergeResult UNSUCCESSFUL = new MergeResult();
+    public static <T> MergeResult<T> unsuccessful() {
```

Comment:
why change this?

Reply:
To make it typesafe (ie. of type T) rather than of type Object.